.cirrus.yml

```diff
@@ -64,12 +64,19 @@ compute_credits_template: &CREDITS_TEMPLATE
   use_compute_credits: $CIRRUS_REPO_FULL_NAME == 'bitcoin/bitcoin' && $CIRRUS_PR != ""
 
 task:
-  name: 'lint [bionic]'
+  name: 'lint [jammy]'
   << : *BASE_TEMPLATE
   container:
-    image: ubuntu:bionic  # For python 3.6, oldest supported version according to doc/dependencies.md
-    cpu: 1
-    memory: 1G
+    image: ubuntu:jammy
+    cpu: 2 # Assign 2 cores and 4G of ram in case .python-version has changed and cirrus has to rebuild python
+    memory: 4G
+  python_cache:
+    folder: /tmp/python-build # Uses intermediary directory for python caching. See the install.sh script in ci/lint/
+    fingerprint_script:
+      - echo $CIRRUS_OS
+      - cat .python-version
+      - cat ./ci/lint_run_all.sh
+      - cat ./ci/lint/*.sh
```

Comment:
Why?

Reply:
If you change the lint scripts (which might change how you interact with the cache), you'd want to be able to test that no?

Reply:
Still trying to understand you, but we don't want to consume resources to rebuild python after every change in linter scripts. What kind of such changes could lead to different python binaries?

Reply:
Consume? I wouldn't say so, 2 cores and 4G of RAM are small amounts nowadays. If you really want to be specific, then I could put the python build scripts into a separate file and cat only that. Or if you really really want to make sure you don't waist any resources, then a separate task which builds python in the Cirrus CI is needed